packages/symbiosis/src/Symbiosis.test.ts

```diff
@@ -1,138 +1,79 @@
+      test(description, async () => {
+        const filter = await bridge(params)
         expect(apply(transaction, filter)).to.be.true
       })
     })
   })
-  describe('should not pass filter with invalid parameters', () => {
-    test('when sourceChainId is incorrect', async () => {
-      const { transaction, tokenAddress, recipient } = controlTransaction
 
-      const filter = await bridge({
-        sourceChainId: 1, // 42161
-        destinationChainId: 5000,
-        tokenAddress,
-        amount: GreaterThanOrEqual(10000n),
-        recipient,
-      })
-      expect(apply(transaction, filter)).to.be.false
-    })
-    test('when bridge contract address is incorrect', async () => {
-      const { transaction, tokenAddress, recipient } = controlTransaction
-
-      const filter = await bridge({
-        sourceChainId: 42161,
-        destinationChainId: 5000,
-        contractAddress: '0x1DCfbC3fA01b2a86bC3a3f43479cCe9E8D438Adc',
-        tokenAddress,
-        amount: GreaterThanOrEqual(10000n),
-        recipient,
-      })
-      expect(apply(transaction, filter)).to.be.false
-    })
-    FAILING_TEST_TRANSACTIONS.forEach((testTransaction) => {
-      test(testTransaction.description, async () => {
-        const {
-          transaction,
-          destinationChainId,
-          tokenAddress,
-          amount,
-          recipient,
-        } = testTransaction
-
-        const filter = await bridge({
-          sourceChainId: transaction.chainId,
-          destinationChainId,
-          tokenAddress,
-          amount: GreaterThanOrEqual(amount),
-          recipient,
-        })
+  describe('should not pass filter with invalid transactions', () => {
+    failingTestCases.forEach((testCase) => {
+      const { transaction, description, params } = testCase
+      test(description, async () => {
+        const filter = await bridge(params)
         expect(apply(transaction, filter)).to.be.false
       })
     })
   })
-  describe('control transaction should pass filter', () => {
-    test(controlTransaction.description, async () => {
-      const {
-        transaction,
-        destinationChainId,
-        tokenAddress,
-        amount,
-        recipient,
-      } = controlTransaction
 
-      const filter = await bridge({
-        sourceChainId: transaction.chainId,
-        destinationChainId,
-        tokenAddress,
-        amount: GreaterThanOrEqual(amount),
-        recipient,
+  describe('should return a valid list of tokens for each supported chain', async () => {
+    const chainIdArray = await getSupportedChainIds()
+    chainIdArray.forEach((chainId) => {
+      test(`for chainId: ${chainId}`, async () => {
+        const tokens = await getSupportedTokenAddresses(chainId)
+        const addressRegex = /^0x[a-fA-F0-9]{40}$/
+        expect(tokens).to.be.an('array')
+        expect(tokens).to.have.length.greaterThan(0)
+        expect(tokens).to.have.length.lessThan(100)
```

Comment:
Is 100 just an arbitrary limit that we set for now?

Reply:
Ya, its arbitrary. Sometimes these sdks try to return massive lists of tokens. For example, uniswaps list has a few thousand, so we ended up just curating a list of popular tokens instead. So the test is basically just to alert us if an oversized list is being provided.

I didnt really put much though into the number used.